### PR TITLE
Add E2E tests to cover breaking changes on secrets and services

### PR DESCRIPTION
**Pull request by sebgl**
Renaming a service, a secret or a secret key is a breaking change since
users may already rely on the naming convention.
Let's catch those in E2E tests.

Fixes https://github.com/elastic/cloud-on-k8s/issues/3036.

### CODE REVIEW

File: test/e2e/test/elasticsearch/checks_k8s.go [hunk 1]
```diff
@@ -72,6 +75,30 @@ func CheckCertificateAuthority(b Builder, k *test.K8sClient) test.Step {
 	}
 }
 
+// CheckSecrets checks that expected secrets have been created.
+func CheckSecrets(b Builder, k *test.K8sClient) test.Step {
+	return test.CheckSecretsContent(k, b.Elasticsearch.Namespace, func() map[string][]string {
```
[the remark this answers is not in the document]
Comment (sebgl) [reply test/e2e/test/elasticsearch/checks_k8s.go:80]: Good idea

File: test/e2e/test/apmserver/checks_k8s.go [hunk 2]
```diff
@@ -123,22 +123,76 @@ func CheckServicesEndpoints(b Builder, k *test.K8sClient) test.Step {
...
 		if b.ApmServer.Spec.ElasticsearchRef.Name != "" {
 			expected = append(expected,
-				test.ExpectedSecret{Name: apmName + "-apm-es-ca", Keys: []string{"ca.crt", "tls.crt"}},
-				test.ExpectedSecret{Name: apmName + "-apm-user", Keys: []string{b.ApmServer.Namespace + "-" + apmName + "-apm-user"}},
+				test.ExpectedSecret{
+					Name: apmName + "-apm-es-ca",
+					Keys: []string{"ca.crt", "tls.crt"},
+					Labels: map[string]string{
+						"apmassociation.k8s.elastic.co/name":        apmName,
+						"apmassociation.k8s.elastic.co/namespace":   apmNamespace,
```
Comment (barkbay) [inline test/e2e/test/apmserver/checks_k8s.go:154]: CA copy also contains `apmassociation.k8s.elastic.co/type: elasticsearch` , not sure if you removed it on purpose ?

Comment (sebgl) [reply test/e2e/test/apmserver/checks_k8s.go:154]: I don't see that label on the resource. Is it something we used to have?

Comment (sebgl) [reply test/e2e/test/apmserver/checks_k8s.go:154]: Ah, is it part of https://github.com/elastic/cloud-on-k8s/pull/3043?

Comment (barkbay) [reply test/e2e/test/apmserver/checks_k8s.go:154]: Sorry, I forgot to add some context here. Let's take this one into consideration later when #3043 is merged.

**Pull request merged**